File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> char_counts;

        for (char c : s) {
            char_counts[c]++;
        }

        int palindrome_length = 0;
        bool odd_found = false;

        for (auto& pair : char_counts) {
            int count = pair.second;
            if (count % 2 == 0) {
                palindrome_length += count;
            } else {
                palindrome_length += count - 1;
                odd_found = true;
            }
        }
        if (odd_found) {
            palindrome_length += 1;
        }

        return palindrome_length;
    }
};

File: 0409-longest-palindrome/0409-longest-palindrome.cs
public class Solution {
    public int LongestPalindrome(string s) {
        
        Dictionary<char, int> charCounts = new Dictionary<char, int>();
        
        foreach (char c in s)
        {
            if (charCounts.ContainsKey(c))
            {
                charCounts[c]++;
            }
            else
            {
                charCounts[c] = 1;
            }
        }
        
        int palindromeLength = 0;
        bool oddFound = false;
        
        foreach (int count in charCounts.Values)
        {
            if (count % 2 == 0)
            {
                palindromeLength += count;
            }
            else
            {
                palindromeLength += count - 1;
                oddFound = true;
            }
        }
        if (oddFound)
        {
            palindromeLength += 1;
        }
        
        return palindromeLength;
    }
}